src/foam/core/stdlib.js

```diff
@@ -361,7 +361,7 @@ foam.LIB({
     function clone(o) { return o; },
     function equals(a, b) { return a === b; },
     function compare(a, b) {
-      return ( b === null || b === undefined ) ? 1 :
+      return ( b === null || b === undefined || foam.typeOf(b).id === 'foam.core.FObject') ? 1 :
```

Comment:
Due to the inversion of control introduced above, when foam.util.compare is called with an object and number, this method will be called. Without this check, we would get 0 since (a !< b && a !> b).